### DISCUSSION TITLE
S22 won't talk with Dex Share (Android 12)

### DISCUSSION THREAD

**Discussion by N3FM** [NightscoutFoundation/xDrip#2061] [Q&A]
Brand new Samsung Galaxy S22+, upgrading (?) from S10+.  (ANDROID 12)
First I tried using Dexcom S6 to grab BG, use Dexcom Share to read into XDrip+ (ver 49817a5-2021.10.04, code 2110041402). Error messages "Get Share Data Failed: java.lang.NumberFormatException: For input string: "Flat"" (or FortyFiveUp, etc.)  (I KNOW this was solved in the past - so what am I doing to see it again?)

Then I tried to use xDrip+ to acquire the data from the G6 (successfully) and then push it to Dexcom Share servers. Now I'm getting "Got exception trying to use Android 10+ service starting for ShareFollowService java.lang.illegalArgumentException: foregroundServiceType 0x00000008 is not a subset of foregroundServiceType attribute 0x00000000 in service element of manifest file." I can see it in xDrip but it doesn't get sent to Dex Share. 

Goal: visible on xDrip+ for nighttime lockscreen, DexShare for concerned others, Fitbit Sense watch... 

Phone is Android 12, kernel  5.10.43-android12-9-abS906USQU1AVC8, build SP1A.210812.016.S906USQU1AVC8

**Comment by N3FM** [comment]
Update: after realizing that clicking on the 'latest' release only got me an October version, I went back in and pulled the 2022.03.28 nightly. I know (at least) the NumberFormatException is resolved.  Newest symptom is that it will not read the G6 when sourcing from the sensor, failing raw bounds validation -0.1486... (this was not happening in 2021.10.04)

**Reply by Navid200** [reply]
The latest release in the channel you are on as explained here:
https://navid200.github.io/xDrip/docs/Updates.html

There has been no change since October 4 affecting collection from G6.
Just restart the phone.

If the problem persists, make sure the settings are correct:
https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html

If they are, post a screenshot of the system status page.  You can blackout the serial number if you like.

**Reply by N3FM** [reply]
Thanks for ALL your help, @Navid200 !  My current issue with raw readings MIGHT be due to aging transmitter - expires in 8 days... so I'll sit on this issue until I replace the transmitter.

**Comment by N3FM** [comment]
Discussion #688 suggested failing transmitter. Confirmed that mine was within 8 days of expiration, so went ahead and swapped it out. Will advise after sensor restart.

**Reply by tolot27** [reply]
It should work if you started the sensor on day 99 or before.
